src/Inventory/Inventory.php

```diff
@@ -847,13 +847,34 @@ public static function cronCleanorphans($task)
         global $DB;
 
         $cron_status = 0;
-
         $conf = new Conf();
-        $existing_types = glob(GLPI_INVENTORY_DIR . '/*', GLOB_ONLYDIR);
+        $now   = time();
+
+        //first, cleanup temporary files
+        $temp_files = new \RegexIterator(
+            new \RecursiveIteratorIterator(
+                new \RecursiveDirectoryIterator(GLPI_INVENTORY_DIR)
+            ),
```

Comment:
Maybe I miss something, but I think here only the temporary files located directly inside `GLPI_INVENTORY_DIR` should be cleaned. Other files will be cleaned by `//clean orphaned inventory files` loop.

```suggestion
            new \DirectoryIterator(GLPI_INVENTORY_DIR),
```

Reply:
Temporary files are only sotred at the main level. But anyway, it's not required to read all subdirectories :)